opencti-platform/opencti-front/src/private/components/trash/all/DeleteOperationPopover.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useState } from 'react';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import MoreVert from '@mui/icons-material/MoreVert';
+import { graphql, useMutation } from 'react-relay';
+import { PopoverProps } from '@mui/material/Popover';
+import IconButton from '@mui/material/IconButton';
+import DialogTitle from '@mui/material/DialogTitle';
+import { DeleteOperationsLinesPaginationQuery$variables } from './__generated__/DeleteOperationsLinesPaginationQuery.graphql';
+import { useFormatter } from '../../../../components/i18n';
+import Transition from '../../../../components/Transition';
+import useDeletion from '../../../../utils/hooks/useDeletion';
+import { RelayError } from '../../../../relay/relayTypes';
+import { MESSAGING$ } from '../../../../relay/environment';
+import { deleteNode } from '../../../../utils/store';
+
+const DeleteOperationPopoverConfirmMutation = graphql`
+  mutation DeleteOperationPopoverConfirmMutation($id: ID!) {
+    deleteOperationConfirm(id: $id)
+  }
+`;
+
+const DeleteOperationPopoverRestoreMutation = graphql`
+  mutation DeleteOperationPopoverRestoreMutation($id: ID!) {
+    deleteOperationRestore(id: $id)
+  }
+`;
+
+interface DeleteOperationPopoverProps {
+  mainEntityId: string;
+  deletedCount: number;
+  disabled?: boolean;
+  paginationOptions: DeleteOperationsLinesPaginationQuery$variables;
+}
+
+const DeleteOperationPopover: React.FC<DeleteOperationPopoverProps> = ({ mainEntityId, deletedCount, disabled, paginationOptions }) => {
+  const { t_i18n } = useFormatter();
+  const [anchorEl, setAnchorEl] = useState<PopoverProps['anchorEl']>();
+  const [commitConfirm] = useMutation(DeleteOperationPopoverConfirmMutation);
+  const [commitRestore] = useMutation(DeleteOperationPopoverRestoreMutation);
+
+  const handleOpen = (event: React.SyntheticEvent) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(undefined);
+  };
+
+  const handleRestore = () => {
+    // TODO: add confirm dialog, though a hook useConfirmDialog() that could replace also useDeletion
```

Comment:
TODO for later?

Reply:
If we have time. There is so much to do if we want to make this right, modern react and all.